### ISSUE TITLE
Feature/support docker '--network' parameter providing userdefined networks (thus dns/autodiscovery for its containers)

### ISSUE THREAD

**Issue by lhoss** [metacloud/molecule#748]
# Issue Type
- Feature request
Feature already asked in issue https://github.com/metacloud/molecule/issues/650#issuecomment-270351777 but  PR (https://github.com/metacloud/molecule/pull/725) was merged for for v1.20, without addressing the '--network' requirement

# Molecule and Ansible details
```
$ ansible --version
ansible 2.2.0.0
$ molecule --version
molecule, version 1.20.0
```

# Desired Behaviour
In many (testing) cases (at least in bigdata), you want to deploy a sample cluster of containers, providing:
* full connectivity between containers, by re-using a customizable docker network (name)
* automatic DNS resolution (so that each container in the same network can connect to another container, through it's container/host-name)
The good news, this functionality is provided out-of-the-box by using docker user-defined networks, see: https://docs.docker.com/engine/userguide/networking/#/user-defined-networks
So we just need to expose additional parameters to molecule.

I propose to fully expose the original docker params, so not to limit the configurability in any way.
[ An alternative (not recommended!), possibly simpler impl. could be adding a new (molecule custom) 'network mode' (like `udf_network`) that would ensure that a hardcoded custom network gets created and used by all containers that have the same setting (`udf_network`) ]

# Actual Behaviour (Bug report only)
Since v1.20 one can change the docker network *mode* ('bridge' being the default), but still not set a custom network, working in that default `bridge` mode.

**Comment by abtreece** [comment]
@lhoss DNS resolution does not work for the default bridge network. If you create a new (non-default) bridge network it will work. The current docker network capabilities in Molecule basically only allow you to attach them to an existing network. Thus in order to use a non-default bridge you would need to manually create a new bridge network and then set `network_mode` to the name of that network.

I thought (briefly) about adding the ability to manage docker networks via molecule, but that significantly increased the scope of the changes I was making.

**Comment by lhoss** [comment]
OK, now got it (working) 👍    with manually creating a docker network, and then setting network_mode to that network.

The 'network_mode' doc  IMO is confusing, if not wrong on following:
> bridge creates a new network stack for the container on the Docker bridge
This suggests that when using 'bridge' already a *new* (non default) docker network is created. Instead this is simply re-using the default one.

Anyway, my next question  (and another small feature request) is, where/how best to add the network creation ?

I found in the Doc, one can configure custom shell cmds to run on the host, BUT unfortunately it seems molecule currently only supports a single *dependency*.
I'ld have expected you could give a list of dependencies (I tried, but failed), or allowing multiple elements, as below (but only the last element was used): 
```
dependency:
  name: shell
  command: 'docker network create --driver bridge isolated_nw'
dependency:
  name: galaxy
  requirements_file: requirements.yml
```

Another option is to run  an initial 'localhost' ansible play (or just `local` tasks), like:

```
  tasks:
# following requires ansible v2.2:
  - name: "Prepare a custom bridge network (to ensure dns/autodiscovery for our docker cluster)"
    docker_network:
      name: "isolated_nw"
```

**Comment by retr0h** [comment]
Implemented in #780.
